pipeline_games/load_games.py

```diff
@@ -0,0 +1,289 @@
+        cur.execute(
+            """SELECT exists (SELECT 1 FROM game_publisher_link 
+            WHERE game_id = %s AND publisher_id = %s LIMIT 1);""",
+            [game_id, publisher_id])
+        result = cur.fetchone()
+        if result['exists'] is True:
+            cur.close()
+        elif result['exists'] is False:
+            cur.execute(
+                """INSERT INTO game_publisher_link(game_id, publisher_id)
+                VALUES (%s, %s);""", [game_id, publisher_id])
+        conn.commit()
+        cur.close()
+
+
+def add_to_developer_link_table(conn: connect, data: list) -> None:
+    """Updates link table"""
+    with conn.cursor(cursor_factory=RealDictCursor) as cur:
+        cur.execute(
+            """SELECT developer_id FROM developer WHERE developer_name = %s;""",
+            [data[10]])
+        developer_id = cur.fetchone()['developer_id']
+        cur.execute(
+            """SELECT game_id FROM game WHERE app_id = %s;""",
+            [data[2]])
+        game_id = cur.fetchone()['game_id']
+        cur.execute(
+            """SELECT exists (SELECT 1 FROM game_developer_link 
+            WHERE game_id = %s AND developer_id = %s LIMIT 1);""",
+            [game_id, developer_id])
+        result = cur.fetchone()
+        if result['exists'] is True:
+            cur.close()
+        elif result['exists'] is False:
+            cur.execute(
+                """INSERT INTO game_developer_link(game_id, developer_id)
+                VALUES (%s, %s);""", [game_id, developer_id])
+        conn.commit()
+        cur.close()
+
+
+def upload_developers():
+    """Uploads new developers"""
+    developers_cache = {}
+    data_frame['developer_id'] = data_frame["developers"].apply(
+        lambda row: get_existing_data(
+            "developer_id", "developer", "developer_name",
+            row, connection, developers_cache))
+
+    developers_data = data_frame[data_frame.developer_id == "None"]
+    new_developers = developers_data['developers']
+
+    execute_batch_columns(connection, new_developers,
+                          'developer', 'developer_name', page_size=100)
+
+
+def upload_publishers():
+    """Uploads new publishers"""
+    publishers_cache = {}
+    data_frame["publisher_id"] = data_frame["publishers"].apply(
+        lambda row: get_existing_data(
+            "publisher_id", "publisher", "publisher_name",
+            row, connection, publishers_cache))
+
+    publishers_data = data_frame[data_frame.publisher_id == "None"]
+    new_publishers = publishers_data['publishers']
+
+    execute_batch_columns(connection, new_publishers,
+                          'publisher', 'publisher_name', page_size=100)
+
+
+def upload_genres():
+    """Uploads new genres"""
+    genres_cache = {}
+    data_frame['genre_id'] = data_frame.apply(
+        lambda row: get_existing_data_for_genre(
+            row['genre'], row['user_generated'], connection, genres_cache), axis=1)
```

Comment:
btw this line (238) might break when dockerised because here connection is taken from unknown location as it is not in this function at all. Also, if you will be using connection typehint from psycopg2.extensions - you will need to rename this just in case to not clash with the imported connection